src/librustc_middle/ty/relate.rs

```diff
@@ -431,6 +436,9 @@ pub fn super_relate_tys<R: TypeRelation<'tcx>>(
             let t = relation.relate(&a_t, &b_t)?;
             match relation.relate(&sz_a, &sz_b) {
                 Ok(sz) => Ok(tcx.mk_ty(ty::Array(t, sz))),
+                // FIXME(lazy_normalization_consts) Implement improved diagnostics for mismatched array
```

Comment:
We should add `FIXME(lazy_normalization_consts)` everywhere we currently have `unimplemented!()`, if these are left for a follow up.